src/api/app/controllers/webui/request_controller.rb

```diff
@@ -292,6 +292,18 @@ def conversation
     @active = 'conversation'
   end
 
+  def build_results
+    redirect_to request_show_path(params[:number], params[:request_action_id]) unless @action[:sprj] || @action[:spkg]
```

Comment:
This is a temporary patch to fix this misbehavour:

![content_of_build_result_when_no_tab](https://user-images.githubusercontent.com/2581944/236167380-09e18325-263a-4bb2-a3ca-3ef7a63b94af.gif)

Reply:
Build results are not available for certain actions such as `add_role`. That's why we have this redirect. Previously we were just rendering the conversation tab in such cases. 

@saraycp, @danidoni , and I have discussed that a better fix would be a refactoring to decouple requests and actions logic.

Reply:
@rubhanazeem just one thing, could you please make this a separate commit with some description. I know your reasoning's for adding this since we talked, but it gets hard to keep track of those changes in the PR.

Reply:
This redirect will be part of other follow-up PRs as well in a similar pattern. The change we are talking about is not a small refactoring. I'd leave the change as part of the main commit because with that refactoring we might as well remove the whole action